File: Backend/Elevate.Models/Challenge/ChallengeModel.cs
using Elevate.Common.Utilities;
using Elevate.Models.Habit;

namespace Elevate.Models.Challenge
{
    public class ChallengeModel
    {
        public required Guid Id { get; set; }

        public required Guid UserId { get; set; }
        public required Guid FriendId { get; set; }

        public required HabitModel Habit { get; set; }

        public ChallengeInviteStatus Status { get; set; } = ChallengeInviteStatus.Pending;

        public DateTime CreatedAt { get; set; } = DateTime.SpecifyKind(DateTimeConverter.UtcToCetTime(DateTime.UtcNow), DateTimeKind.Utc);
        public DateTime? UpdatedAt { get; set; } = DateTime.SpecifyKind(DateTimeConverter.UtcToCetTime(DateTime.UtcNow), DateTimeKind.Utc);
    }
}

File: Mobile-frontend/src/app/.models/challenge.model.ts
import { Habit } from './Habit.model';
import { User } from './user.model';


export interface Challenge {
    id: string;
    userId: string;
    friendId: string;
    habit: Habit;
    status: string;
    createdAt: string;
    updatedAt: string;
    user?: User;
}